RELEASE.md

```diff
@@ -0,0 +1,135 @@
+
+### Release notes
+
+When working on pre-release PR the person doing the release should check that merged PRs have release notes when necessary.
+
+It can be checked by running a diff with previous release and looking at the PR commits:
+
+```
+git log 0.106.0..HEAD --invert-grep --grep='deps' --grep='docs' --grep='build' | grep -oh '^.*\(#[0-9]*\)'
+```
+
+And verifying that PRs have release notes.
+
+This is a bit tedious, but should be less of an issue as we establish a process of adding release notes.
+
+
+## Handling Failures
+
+### Retry
+
+Depending on where the release failed, some of the following steps may have succeeded:
+
+- Creation of a new tag in the kanister repo
+- PR raised in kanister to update kanister tools
+
+Before rerunning the release job manually, do the following:
+
+- Delete the tag manually from GitHub
+- Close the PRs created (to avoid confusion)
+
+## Manual Release (Only needed when automation is broken)
+
+Prerequisites:
+
+- Make sure you have admin access or access to push tags on Kanister
+- Make sure you have access to [push images to GHCR](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry)
+
+### Create pre-release
+
+If pre-release pipeline does not work.
+
+```
+$ export PREV_TAG="0.42.0"
+$ git checkout -b "kan-docs-${RELEASE_TAG}"
+$ ./build/bump_version.sh "${PREV_TAG}" "${RELEASE_TAG}"
+$ make reno-report VERSION="${RELEASE_TAG}"
+$ git add -A
+$ git commit -m"pre-release: Update version to ${RELEASE_TAG}"
+$ git push origin kan-docs-${RELEASE_TAG}
+```
+
+Create PR from this branch.
+
+### Create release binaries and images
+
+If release pipeline does not work
+
+1. Make sure to merge pre-release PR first.
+
+2. Create a release tag
+
+```
+$ export RELEASE_TAG="0.43.0"
+$ git checkout master
+$ git pull origin master
+$ git tag -a "${RELEASE_TAG}" -m "Release version";
+$ git push origin "${RELEASE_TAG}"
+```
+
+3. Build helm charts
+
+```
+$ export PACKAGE_FOLDER=helm_package
+$ export HELM_RELEASE_REPO_URL=https://github.com/kanisterio/kanister/releases/download/${RELEASE_TAG}
+$ export HELM_RELEASE_REPO_INDEX=https://charts.kanister.io/
+$ make package-helm VERSION=${RELEASE_TAG}
+```
+
+4. Release binaries and docker images
+
+```
+$ make gorelease CHANGELOG_FILE=./CHANGELOG.md GORELEASE_PARAMS='--draft'
+```
+
+5. Update and release docs and helms charts
+
+**Currently Github pages publishing is only supported via Github actions**
+
+### Publish Kanister Release
+
+Finally, go to https://github.com/kanisterio/kanister/releases, and publish the draft release.
+
+## Post-release Checks
+
+- Verify the Kanister [repo](https://github.com/kanisterio/kanister/releases) for the new release tag.
+
+- Verify if the docker [images](https://github.com/orgs/kanisterio/packages?repo_name=kanister) have a new tag. NOTE: Not all docker images are relevant.
+	TODO: Add a list of the most relevant docker images to be verified here.
+
```

Comment:
Should we add another point here about updating the helm repo and checking that new version of chart is updated?

Reply:
Added.